src/App.js

```diff
@@ -1,25 +1,151 @@
-import logo from './logo.svg';
-import './App.css';
+import React, { useState } from "react";
+import Boxes from "./Components/Boxes";
+import "./App.css";
+import ColourPicker from "./Components/ColourPicker";
+
+const App = () => {
+  const [boxes, updateBoxes] = useState([
+    { id: 1, highlight: false, colour: "yellow" },
+    { id: 2, highlight: false, colour: "yellow" },
+    { id: 3, highlight: false, colour: "yellow" },
+    { id: 4, highlight: false, colour: "yellow" },
+    { id: 5, highlight: false, colour: "yellow" },
+    { id: 6, highlight: false, colour: "yellow" },
+    { id: 7, highlight: false, colour: "yellow" },
+    { id: 8, highlight: false, colour: "yellow" },
+    { id: 9, highlight: false, colour: "yellow" },
+  ]);
+
+  const [selectedColour, updateSelectedColour] = useState("yellow");
+
+  const handleHighlight = (id, selectedColour) => {
+    let newBoxes = [...boxes];
+    const boxIndex = id - 1;
+    newBoxes[boxIndex].highlight = !newBoxes[boxIndex].highlight;
+    newBoxes[boxIndex].colour = selectedColour;
+    newBoxes = changeSurrounding(newBoxes, boxIndex, selectedColour);
+    updateBoxes(newBoxes);
+  };
+
+  const handleColourChange = (currentColour, selected) => {
+    let newColour = selected;
+    if (currentColour === newColour) return;
```

Comment:
you could make this shorter by putting

```
if (currentColor !== newColour) updateselectedColour(newColour)
```
This would mean you dont need a return on fail, it just will still not do anything if they match